components/Shell.tsx

```diff
@@ -90,20 +112,10 @@ export default function Shell(props: {
   CTA?: ReactNode;
 }) {
   const router = useRouter();
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
   useRedirectToLoginIfUnauthenticated();
+  useRedirectToOnboardingIfNeeded();
```

Comment:
Extracted onboarding logic to hook. We don't need to use it in every page anymore.